bodhi/tests/client/test_bindings.py

```diff
@@ -24,6 +25,435 @@
+
+        result = client.query(package='bodhi-2.4.0-1.fc26')
+
+        self.assertEqual(result, 'return_value')
+        client.send_request.assert_called_once_with(
+            'updates/', verb='GET', params={'builds': 'bodhi-2.4.0-1.fc26'})
+
+    def test_with_package_fedora_id(self):
+        """
+        Test with the package arg expressed as a Fedora update id.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+
+        result = client.query(package='FEDORA-2017-52506b30d4')
+
+        self.assertEqual(result, 'return_value')
+        client.send_request.assert_called_once_with(
+            'updates/', verb='GET', params={'updateid': 'FEDORA-2017-52506b30d4'})
+
+    def test_with_package_name(self):
+        """
+        Test with the package arg expressed as a package name.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+
+        result = client.query(package='bodhi')
+
+        self.assertEqual(result, 'return_value')
+        client.send_request.assert_called_once_with(
+            'updates/', verb='GET', params={'packages': 'bodhi'})
+
+    def test_with_release(self):
+        """
+        Test with a 'release' kwarg.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+
+        result = client.query(packages='bodhi', release='f26')
+
+        self.assertEqual(result, 'return_value')
+        client.send_request.assert_called_once_with(
+            'updates/', verb='GET', params={'packages': 'bodhi', 'releases': 'f26'})
+
+    def test_with_type_(self):
+        """
+        Test with the type_ kwarg.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+
+        result = client.query(packages='bodhi', type_='security')
+
+        self.assertEqual(result, 'return_value')
+        client.send_request.assert_called_once_with(
+            'updates/', verb='GET', params={'packages': 'bodhi', 'type': 'security'})
+
+
+class TestBodhiClient_save(unittest.TestCase):
+    """
+    This class contains tests for BodhiClient.save().
+    """
+    def test_with_type_(self):
+        """
+        Assert that save() handles type_ as a kwargs for backwards compatibility.
+        """
+        client = bindings.BodhiClient()
+        client.send_request = mock.MagicMock(return_value='return_value')
+        client.csrf_token = 'a token'
+        kwargs = {
+            'builds': ['bodhi-2.4.0-1.fc26'], 'type_': 'enhancement', 'notes': 'This is a test',
+            'request': 'testing', 'autokarma': True, 'stable_karma': 3, 'unstable_karma': -3,
+            'severity': 'low'}
+
+        response = client.save(**kwargs)
+
+        self.assertEqual(response, 'return_value')
+        kwargs['type'] = kwargs['type_']
```

Comment:
The request got sent two lines above in the call to ```client.save()```. This line is just adjusting the kwargs for the purpose of asserting that ```save()``` properly modified the kwargs.